src/components/Form/FormSection.tsx

```diff
@@ -24,7 +24,7 @@ export interface FormSectionProps<TData extends GenericFormValues, TErrors> {
   saveButton?: string;
   cancelButton?: string;
   canSave?: (values: TData) => boolean;
-  onSubmit?: (values: TData) => Promise<{ errors?: string[] }>;
+  onSubmit?: (values: TData) => Promise<{ errors?: string[] }> | Promise<void> | void;
```

Comment:
We can't add this back in. The voids can too easily cause problems with dropped responses (there was a bug where errors from the email update weren't getting handled)
```suggestion
  onSubmit?: (values: TData) => Promise<{ errors?: string[] }>;
```

Reply:
also done